forge-game/src/main/java/forge/game/GameAction.java

```diff
@@ -1316,7 +1317,11 @@ else if (c.hasBeenDealtDeathtouchDamage() || (c.getDamage() > 0 && c.getLethal()
                         }
                         desCreats.add(c);
                         c.setHasBeenDealtDeathtouchDamage(false);
-                        checkAgainCard = true;
+                        boolean check = true;
+                        if (game.getPhaseHandler() != null && c.hasKeyword("Damage isn't removed from CARDNAME during cleanup steps.")) {
+                            check = PhaseType.CLEANUP != game.getPhaseHandler().getPhase();
+                        }
+                        checkAgainCard = check;
```

Comment:
Hmn, I‘m not sure force setting this to false couldn‘t cause issues with other cards of that run rules wise
Gonna debug tonight

Reply:
I would suggest having an else if line on 1302 that does nothing if the creature has indestructible. Then move on to see if the creature was destroed from damage-y things.

Reply:
yes good call:
> 702.12b. A permanent with indestructible can't be destroyed. Such permanents aren't destroyed by lethal damage, **and they ignore the state-based action that checks for lethal damage**

This was actually pretty bad for performance because it meant indestructibles with lethal would cause the loop to always run the full 9 iterations on each SBA step (until cleanup)